colossalai/tensor/_ops/linear.py

```diff
@@ -7,7 +7,7 @@
 from colossalai.core import global_context as gpc
 from packaging import version
 from colossalai.utils.cuda import get_current_device
-
+from colossalai.tensor.spec import TensorSpec, ComputePattern, ParallelAction
```

Comment:
from colossalai.tensor import
Use the public API in shallow directory as much as possible

Reply:
Done